lib/msf/core/post/file.rb

```diff
@@ -506,14 +506,15 @@ def rm_rf(*remote_dirs)
   # @param old_file [String] Remote file name to move
   # @param new_file [String] The new name for the remote file
   def rename_file(old_file, new_file)
+    raise "File to be renamed doesn't exists" unless exist?(old_file)
     if session.type == "meterpreter"
       return (session.fs.file.mv(old_file, new_file).result == 0)
+    elsif session.type == 'powershell'
+      cmd_exec("Rename-Item -Path #{old_file} -NewName #{new_file}")
```

Comment:
Will this work if there are spaces in the path? It looks like the arguments may need to be wrapped in double quotes like the others, but I haven't tested it.

Reply:
Yeah, I forgot to double quotes, will add them.